llava/train/llava_trainer.py

```diff
@@ -849,4 +794,84 @@ def _inner_training_loop(
         if self.neftune_noise_alpha is not None:
             self._deactivate_neftune(self.model)
 
-        return TrainOutput(self.state.global_step, train_loss, metrics)
\ No newline at end of file
+        return TrainOutput(self.state.global_step, train_loss, metrics)
+    
+    def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor, Any]], dmode: bool = False) -> torch.Tensor:
+
+        inputs['d_mode'] = dmode
+
+        if not dmode: 
+            for name, param in model.named_parameters(): 
+                if "discriminator" in name: 
+                    param.requires_grad = False
+        
+        if dmode: 
+            for name, param in model.named_parameters(): 
+                if "discriminator" in name: 
+                    param.requires_grad = True
```

Comment:
ensures mutually exclusive updates 

the optimizer was fixed and pushed on main in this pr: 
https://github.com/sophiamirrashidi/LLaVAFork/commit/cd499f559df007774f6f99e21b1e63b06577cf9c